libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3217,7 +3219,7 @@ Declaration const& TypeChecker::dereference(UserDefinedTypeName const& _typeName
 bool TypeChecker::expectType(Expression const& _expression, Type const& _expectedType)
 {
 	_expression.accept(*this);
-	if (!type(_expression)->isImplicitlyConvertibleTo(_expectedType))
+	if (BoolResult result = type(_expression)->isImplicitlyConvertibleTo(_expectedType); !result)
```

Comment:
```suggestion
	if (
		BoolResult result = type(_expression)->isImplicitlyConvertibleTo(_expectedType);
		!result
	)
```

Reply:
I would actually find it extremely much more readable by pulling out `result` to the function scope.

Reply:
I can do it, the file uses both forms.